crates/miden-testing/src/kernel_tests/tx/test_faucet.rs

```diff
@@ -529,32 +529,59 @@ fn test_burn_non_fungible_asset_succeeds() {
         begin
             exec.prologue::prepare_transaction
 
-            # add existing non-fungible asset to the vault
+            # add non-fungible asset to the vault
             exec.memory::get_input_vault_root_ptr push.{non_fungible_asset}
             exec.asset_vault::add_non_fungible_asset dropw
```

Comment:
I know this is not your code, but I think that this is an implementation detail of the transaction kernel that a test shouldn't have to setup. It'd be better to have the asset be part of the faucet account's vault, then remove it from there and burn it. But I don't really like any of the tests that use the `TransactionContextBuilder` for setup. I think it makes it fairly hard to understand the initial state of the test and we should refactor that at some point, so let's leave it as-is for now.

Reply:
Should we create an issue for this?